src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -296,6 +323,87 @@ private ODataResourceSet ReadFeedCore(bool lazy)
             return result;
         }
 
+        /// <summary>
+        /// Reads an <see cref="ODataDeltaResourceSet"/>.
+        /// </summary>
+        /// <returns>The <see cref="MaterializerDeltaFeed"/> of the read <see cref="ODataDeltaResourceSet"/>.</returns>
+        private MaterializerDeltaFeed ReadDeltaFeedCore()
+        {
+            this.ExpectState(ODataReaderState.DeltaResourceSetStart);
+
+            ODataDeltaResourceSet result = (ODataDeltaResourceSet)this.reader.Item;
+            MaterializerDeltaFeed feed = null;
+
+            if (result == null)
+            {
+                return null;
+            }
+
+            feed = MaterializerDeltaFeed.CreateDeltaFeed(
+                    result,
+                    new List<IMaterializerState>(),
+                    this.materializerContext);
+
+            if (this.items?.Count > 0)

```

Comment:
Maybe add a comment here explaining why it's important/necessary to have at least one item in the collection before pushing the `feed` into the stack